Modules/mathmodule.c

```diff
@@ -2016,6 +2016,78 @@ math_factorial(PyObject *module, PyObject *arg)
 }
 
 
+/*[clinic input]
+math.lcm
+
+	x as a: object
+	y as b: object
+    /
+least common multiple of x and y
+[clinic start generated code]*/
+
+static PyObject*
+math_lcm(PyObject*module,PyObject*a,PyObject*b)
+{
+    PyObject * g,*m,*f,*ab;
+    if (PyFloat_Check(a)||PyFloat_Check(b)) {
+    if (PyErr_WarnEx(PyExc_DeprecationWarning,
+                    "lcm() cannot be used with float",
+                    1) < 0)
+        {
+            return NULL;
+        }
+    double dx = PyFloat_AS_DOUBLE((PyFloatObject *)a);
+    double ax=   PyFloat_AS_DOUBLE((PyFloatObject *)b);
+    if (!((Py_IS_FINITE(dx) && dx == floor(dx))||(Py_IS_FINITE(ax) && ax == floor(ax))) 
+		{
+            PyErr_SetString(PyExc_ValueError,
+                            "lcm() only accepts integral values");
+            return NULL;
+        }
+    a=PyNumber_Index(a);
+    if(a==NULL)
+        return NULL;
```

Comment:
This code should be reformatted to respect PEP 7 code style. See other C code in the same file, or in other C files around.